docs/DeveloperGuide.adoc

```diff
@@ -241,7 +339,95 @@ image::CommitActivityDiagram.png[]
+
+Step 2. The user fills in the form and presses `f2` on the keyboard or clicks the `submit` button to save the report. This calls `VisitRecordWindow#saveReport()` which will create a new `SaveVisitCommand` and executes it, updating the `VisitList` of the person via `Model#setPerson()`.
+
+Step 3. The user decides to edit the report and executes the `editvisit 1 i/1` command.`MainWindow#executeCommand()` is called and the check for `CommandResult#isEditVisit()` passes, calling  `VisitRecordWindow#setOldReportInfo()` which initializes the respective attributes of `VisitRecordWindow` and populate the form with the content of the `VisitReport` from  `CommandResult#getOldReport()` .`MainWindow#handleShowVisitForm()` is then called to display a pop up form for the user to fill in.
+
+image::EditVisit0.png[]
+
+[NOTE]
+If the user executes `editvisit 1` instead, a `CommandResult` with a `ObservableList<VisitReport>` object will be returned. `MainWindow#executeCommand()` runs and the check for `CommandResult#isEditVisit()` fails while the check for `CommandResult#isShowVisitList()` passes, calling `VisitListPanel#setup()` which populates the `VisitListPanel` with the contents of the `ObservableList<VisitReport>` object from `CommandResult#getObservableVisitList()`.`MainWindow#handleShowVisitList()` is then called to display a pop up list showing an indexed list of all visitation records of the person.
+
+Step 4. The user edits the form and presses `f2` on the keyboard or clicks the `submit` button to save the changes. The same mechanism mentioned in `Step 2` is used to update the contents of the edited visit report.
+
+Step 5. The user then decides that the report was beyond salvaging and decides to delete it. Executing the command `deletevisit 1 d/1`, `VisitList#deleteRecord()` deletes the entry from the arraylist and `Model#setPerson()` updates the person with the new `VisitList`. The `CommandResult#isShowVisitList()` check in  `MainWindow#executeCommand()` passes and a pop up list showing an indexed list of all visitation records of the person is displayed.
+
+image::DeleteVisit0.png[]
+
+[NOTE]
+If the user executes `deletevisit 1` instead, the index value will be set to `-1` and the `VisitList#deleteRecord()` block would be skipped, but the rest of the flow remains the same.
+
+
+==== Design Considerations
+
+===== Aspect: How SaveVisitCommand/DeleteVisitCommand executes
+
+* Creates a new `Person` object each time and using `Model#setPerson()` to replace the original person for every change in `VisitList `
+** Pros: Easy to implement and reuses existing code.
+** Cons: May have performance issues in terms of memory usage.
+
+
+===== Aspect: Data structure to support the addvisit/deletevisit/editvisit commands
+
+*  Use a arraylist to store the `VisitReport` objects in `VisitList` object
+** Pros: Fast retrieval by index.
+** Cons: Need to handle index out of bound errors.
```

Comment:
Try to explain the alternatives besides these design that you have chosen and why your current choice is better than the alternatives!